### PR TITLE
NumberBox: the value resets after "-" is clicked if format is "0.00" (T1092593)

### PR DESCRIPTION
**Pull request by EugeniyKiyashko**

### CODE REVIEW

File: testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js [hunk 1]
```diff
@@ -440,6 +440,21 @@ QUnit.module('format: sign and minus button', moduleConfig, () => {
         assert.deepEqual(this.keyboard.caret(), { start: 3, end: 4 }, 'caret preserved');
     });
 
+    QUnit.test('NumberBox should process a value correctly after valueChange event (T1092593)', function(assert) {
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js]: this name is too common for me

File: testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js [hunk 2]
```diff
@@ -440,6 +440,21 @@ QUnit.module('format: sign and minus button', moduleConfig, () => {
...
+            format: '0.00',
+        });
+
+        this.input.focus();
+        this.keyboard
+            .type('5')
+            .type('-')
+            .change();
+
+        assert.equal(this.input.val(), '-5.00', 'value is correct');
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js]: we try to use `strictEqual` everywhere

File: js/ui/number_box/number_box.mask.js [hunk 3]
```diff
@@ -302,7 +302,6 @@ const NumberBoxMask = NumberBoxBase.inherit({
             }
         }
 
-        text = this._removeStubs(text, true);
```
Comment (ksercs) [inline js/ui/number_box/number_box.mask.js:305]: seems like this function expects full text with stubs as a parameter. Are u sure that 298 row works correctly now? I think that `text.length` can return unexpected value now

File: testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js (hunk 2 above)
Comment (EugeniyKiyashko) [reply testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js]: 3786f2da4

File: testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js (hunk 1 above)
Comment (EugeniyKiyashko) [reply testing/tests/DevExpress.ui.widgets.editors/numberBoxParts/mask.tests.js]: 3786f2da4

File: js/ui/number_box/number_box.mask.js (hunk 3 above)
Comment (EugeniyKiyashko) [reply js/ui/number_box/number_box.mask.js:305]: No tests are failed. May be, do you have a sample that demonstrate incorrect case?

Comment (Zedwag) [reply js/ui/number_box/number_box.mask.js:305]: this code was added for T972648, please test scenarios from this ticket

Comment (Zedwag) [reply js/ui/number_box/number_box.mask.js:305]: https://teams.microsoft.com/l/message/19:82ff4bea5a9f44659092e290f5f90bbe@thread.skype/1613164258551?tenantId=e4d60396-9352-4ae8-b84c-e69244584fa4&groupId=8c31b3e5-5870-46b1-92a7-180d13b282ed&parentMessageId=1613164258551&teamName=DevExtreme&channelName=Editors%20Squad&createdTime=1613164258551

Comment (EugeniyKiyashko) [reply js/ui/number_box/number_box.mask.js:305]: I checked these scenarios. They work as expected.

**Review by Zedwag — DISMISSED** [review]

**Review by Zedwag — APPROVED** [review]

**Review by ksercs — APPROVED** [review]

**Pull request merged**